lib/manager/maven/readme.md

```diff
@@ -0,0 +1,138 @@
+## Overview
+
+#### Name of package manager
+
+Maven
+
+---
+
+#### What language does this support?
+
+Java
+
+---
+
+#### Does that language have other (competing?) package managers?
+
+Gradle 
+
+## Package File Detection
+
+#### What type of package files and names does it use?
+
+"pomfile" usually named pom.xml. It is possible to override default pom.xml with any filename
+
+ `mvn -f <path>/custom_pom.xml <command>`
+
+---
+
+#### What [fileMatch](https://renovatebot.com/docs/configuration-options/#filematch) pattern(s) should be used?
+
+`['(^|/)([\\w-]*)pom.xml$']`
+
+---
+
+#### Is it likely that many users would need to extend this pattern for custom file names?
+
+Not many but some user might use custom filenames.
+
+---
+
+#### Is the fileMatch pattern likely to get many "false hits" for files that have nothing to do with package management?
+
+Usually "pomfile" is in the root of the project and it is not likely that it will be many other xml files in the same directory.
+
+## Parsing and Extraction
+
+#### Can package files have "local" links to each other that need to be resolved?
+
+No
+
+#### Is there reason why package files need to be parsed together (in serial) instead of independently?
+
+No
+
+---
+
+#### What format/syntax is the package file in? e.g. JSON, TOML, custom?
+
+XML
+
+---
+
+#### How do you suggest parsing the file? Using an off-the-shelf parser, using regex, or can it be custom-parsed line by line?
+
+Any XML reader.
+
+---
+
+#### Does the package file structure distinguish between different "types" of dependencies? e.g. production dependencies, dev dependencies, etc?
+
+No
+
+---
+
+#### List all the sources/syntaxes of dependencies that can be extracted:
+
+---
+
+#### Describe which types of dependencies above are supported and which will be implemented in future:
+
+## Versioning
+
+#### What versioning scheme do the package files use?
+
+---
+
+#### Does this versioning scheme support range constraints, e.g. `^1.0.0` or `1.x`?
+
+No, it has its own syntax for supporting [versions](https://maven.apache.org/enforcer/enforcer-rules/versionRanges.html).
+
+---
+
+#### Is this package manager used for applications, libraries, or both? If both, is there a way to tell which is which?
+It is general package manager. No need to separate it.
```

Comment:
It is important if we are able to determine which projects are applications vs which are libraries, as we recommend different behaviour in each circumstance.

Reply:
I meant, we want to be able to determine if the current working repository is for an application, or for a library? I am assuming that people build libraries with Maven dependencies too, not just applications?

Reply:
sure
but I don't really see the difference between application and library in this context

Reply:
Libraries have a need to keep dependency ranges as wide as possible (to maximise compatibility with other libraries in the build) while applications should lock down their dependencies to a single value to reduce the chance of having different versions running on different machines.